flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -646,10 +653,14 @@ protected void cleanUpInvoke() throws Exception {
 		try {
 			channelIOExecutor.shutdown();
 		} catch (Throwable t) {
-			LOG.error("Error during shutdown the channel state unspill executor", t);
+			suppressedThrowable = ExceptionUtils.firstOrSuppressed(t, suppressedThrowable);
 		}
 
 		mailboxProcessor.close();
+
+		if (suppressedThrowable != null) {
+			throw (Exception) suppressedThrowable;
```

Comment:
I think `suppressedThrowable` doesn't have to be `Exception`.

Reply:
It does not, but have to be transmitted to Exception either here or in the function that caught it (outside this function).

Reply:
So `OutOfMemoryError` will be masked as `ClassCastException` at the runtime which is not desirable.

Reply:
resolved.